### PR TITLE
Update section regarding commit messages

### PR DESCRIPTION
**Pull request by johngriebel**
Do issue #6.
Signed-off-by: John Griebel <johnkgriebel@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,13 +37,35 @@ In your pull request description, please include the issue number related to you
...
+```
+
+`type` comes from the issue linked by `issuenumber` (note that this implicitly requires a corresponding issue to exist before any PR can be merged), and is one of:
+
+* chore
+* doc 
+    - Updating documentation to reflect behavioral changes. Primarily used in the openapi repo.
+* epic 
+    - An issue that encompasses multiple issues required to complete a large issue. The epic label should rarely, if ever, be used for a pull request.
+* feature
```
Comment (raygesualdo) [inline CONTRIBUTING.md]: Conventional Commits shortens this to `feat`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,13 +37,35 @@ In your pull request description, please include the issue number related to you
...
+* fix
+* prototype
```
Comment (raygesualdo) [inline CONTRIBUTING.md]: If `feature` is shortened to `feat`, should we shorten this to `proto`? Might be overkill.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -37,13 +37,35 @@ In your pull request description, please include the issue number related to you
...
-- `doc`: If the commit is a change to documentation.
-- `test`: If the commit contains a change to the test suite.
+The only hard and fast rules for commit messages pertain to merge commits themselves, 
+which are done by core contributors. Merge commits follow the [conventional commits](https://conventionalcommits.org) guidelines.
+
+Specifically, 3Blades uses the following format:
+
+```
+<type>: issue #<issuenumber> - <description>
+[optional body]
```
Comment (jgwerner) [inline CONTRIBUTING.md]: Could you provide a full example in the doc? This may help with new contributor on-boarding. Also, by convention, the body should probably be separated by a new line. So the commit message could be structured like so: 

    git commit -s -m $'doc: issue #100\n\nupdate readme with bullets...'

Comment (johngriebel) [reply CONTRIBUTING.md]: Sure, I'll add an example. That's a good idea. Keep in mind this is only for merge commits, so Github will take care of the body being on a new line since it's all in an HTML text box anyhow.

Comment (jgwerner) [reply CONTRIBUTING.md]: I see. Thanks for clarifying.
